crates/typst/src/text/mod.rs

```diff
@@ -463,6 +462,49 @@ pub struct TextElem {
     #[ghost]
     pub hyphenate: Hyphenate,
 
+    /// The "cost" of hyphenation when laying out text.
+    /// A higher cost means the layout engine will hyphenate less often.
+    /// Costs are specified as a ratio of the default cost,
+    /// so `50%` will make text layout twice as eager to hyphenate,
+    /// while `200%` (or 2) will make it half as eager.
+    ///
+    /// Hyphenation is generally avoided by placing the whole word on the next line,
+    /// so a higher runt cost can result in awkward justification spacing.
+    ///
+    /// The default cost is an acceptable balance,
+    /// but some may find that it hyphenates too eagerly,
+    /// breaking the flow of dense prose.
+    /// A cost of 6 (six times the normal cost) may work better for such contexts.
+    pub hyphenation_cost: Option<Ratio>,
+
+    /// The "cost" of runts (lines with a single word) when laying out text.
+    /// A higher cost means the layout engine will try harder to avoid runts.
+    /// Costs are specified as a ratio of the default cost,
+    /// so `50%` will make text layout try half as hard to avoid runts,
+    /// while `200%` (or 2) will make it try twice as hard.
+    ///
+    /// Runts are avoided by placing more or fewer words on previous lines,
+    /// so a higher runt cost can result in more awkward in justification spacing.
+    ///
+    /// The default cost is an acceptable balance,
+    /// but some may find that it produces too many runts,
+    /// which some style guides strongly discourage.
+    /// A cost of 6 (six times the normal cost) may work better for such contexts.
+    pub runt_cost: Option<Ratio>,
```

Comment:
I can see hyphenation-cost as being useful, but this feels too niche tbh. There are even more costs we could expose, but at some point it's too much.

Reply:
I suppose, but at least in its current state the only non-special costs that the algorithm considers are these two, so I figured they should be exposed.

Reply:
why do you mean with "non-special"?

Reply:
The option also has use case in Chinese typesetting. In [CLREQ Section 3.4.4](https://www.w3.org/TR/clreq/#adjustments_of_orphans_and_widows):

> In the tradition of Chinese composition, an orphan does not make a line, nor does a widow make a page.

"orphan" here has the same meaning as runts in Typst. Chinese users can increase the runt cost to a very large number to follow the CLREQ rules.